### ISSUE TITLE
Create the 1st release version

### ISSUE THREAD

**Issue by Peilin-Yang** [castorini/Anserini#237]
Q: There are some components that I am not quite familiar with and I think it is a good time to refresh the repo? For example, 

- [Speech?](https://github.com/castorini/Anserini/tree/master/src/main/js). Is this component more suitable to be put into another repo? It seems it is pretty independent and seems to be more fit for Castor.  

- [Word Embedding](https://github.com/castorini/Anserini/blob/master/docs/embeddings.md), People have adopted PyTorch and put DL4J into desert? 

- [YoGosling](https://github.com/castorini/Anserini/blob/master/docs/yogosling.md). What about this one?

- Anything else?

**Comment by lintool** [comment]
@Peilin-Yang #238 removes the `rts` and `nrts` packages.

**Comment by lintool** [comment]
@Peilin-Yang Can we remove:
+ https://github.com/castorini/Anserini/blob/master/docs/add-collection-class.md
+ https://github.com/castorini/Anserini/blob/master/docs/add-topic-reader.md

The documentation will always be out of date... so we should just make sure code is clear and having good javadocs?

**Comment by Peilin-Yang** [comment]
I would argue that in general lacking of documentation would make future developers on-boarding harder...
But I also agree that for this project it might be the case that keeping the javadocs up-to-date is easier and more realistic.

**Comment by lintool** [comment]
@Peilin-Yang What about `io.anserini.eval`? My inclination is to use `trec_eval` and `gdeval.pl` and all the standard eval scripts that other in the community use. Subtle eval corner cases crop up now and then (e.g., `trec_eval` tie breaking behavior).

Philosophically, I think Anserini should be opinionated since we are providing repeatable baselines - i.e., offer the "one true way" to do things, instead of multiple (maybe slightly different) ways.

**Comment by Peilin-Yang** [comment]
> Philosophically, I think Anserini should be opinionated since we are providing repeatable baselines - i.e., offer the "one true way" to do things, instead of multiple (maybe slightly different) ways.

Interestingly, I agree with the philosophy while can hardly agree with the proposal:
`trec_eval` requires gcc to be compiled - this largely prevents it from being used on Windows and some of the macOS and even on some of the Linux machines (I've encountered gcc/g++ incompatibility issues on the Linux where I am not the admin). 
And `gdeval.pl` has the similar problem.

I think Anserini should be capable for everyone to reproduce the same results while is agnostic to the running environment? Actually mvn (or even Java?) might be a potential blocker here.... 
I am thinking a Docker image might be good then all the dependencies and evaluation scripts are no longer problems.

**Comment by lintool** [comment]
Okay, in this case, we should switch to using our Java eval tools are part of the regression pipeline? Tools that aren't use will not be maintained and go stale. So this would suggest we switch to using `io.anserini.eval` to generate docs, but behind the scenes our regression checks their values against `trec_eval`, etc.?

**Comment by Peilin-Yang** [comment]
My proposal is to keep both `trec_eval`/`gdeval.pl` and `io.anserini.eval`. I will do the regression tests and see if there is any discrepancy among them.
Meanwhile, I will create a Docker image for Anserini. All the binaries (including IndexCollection, trec_eval, etc) will be exposed. 
sg?

**Comment by lintool** [comment]
> My proposal is to keep both `trec_eval/gdeval.pl` and `io.anserini.eval`. I will do the regression tests and see if there is any discrepancy among them.

If the results are the same we can switch to using `io.anserini.eval` in our official docs.

Re: Docker - I think we should hold off... there are probably more urgent tasks to work on...

**Comment by lintool** [comment]
@Peilin-Yang Very soon we'll have repeatable regression tests for the following:

```
nohup python src/main/python/run_regression_disks12.py >& log.disks12 &
nohup python src/main/python/run_regression_robust04.py >& log.robust04 &
nohup python src/main/python/run_regression_robust05.py >& log.robust05 &

nohup python src/main/python/run_regression_wt10g.py >& log.wt10g &
nohup python src/main/python/run_regression_gov2.py >& log.gov2 &
nohup python src/main/python/run_regression_clueweb09b.py >& log.clueweb09b &
nohup python src/main/python/run_regression_clueweb12-b13.py > log.clueweb12-b13 &

nohup python src/main/python/run_regression_tweets2011.py >& log.tweets2011 &
nohup python src/main/python/run_regression_tweets2013.py >& log.tweets2013 &
```

Shall we create an official Anserini v0.1.0 release?

**Comment by Peilin-Yang** [comment]
Yea, I cannot wait for this.
But what about starting the version from `1.0.0`?

**Comment by lintool** [comment]
`1.0.0` implies that Anserini is mature... I don't think we're at that stage yet... I'm thinking `0.1.0`, `0.2.0`, etc. We get a few releases, and then jump to `1.0.0`.

Before `1.0.0` I'd like to have better testing, good Javadocs, etc.

**Comment by Peilin-Yang** [comment]
I think we should wait for the fix #270 before the release?

**Comment by lintool** [comment]
We've release v0.1.0!
